walkingastro.lic

```diff
@@ -69,15 +80,51 @@ class WalkingAstro
 
   def check_predict
     return if DRSkill.getxp('Astrology') > 25
-
     DRCMM.align(@astrology_prediction_skills_survival)
     waitrt?
     DRC.bput('predict future', 'roundtime')
     waitrt?
   end
 
+  def moved_rooms?
+    echo "Checking if you've moved rooms..." if $debug_mode_walkingastro
+    # Check if the room id or room title has changed.
+    # Not all rooms have ids, so we also check the titles.
+    # All rooms have titles, but they may be reused in multiple rooms.
+    # Checking both the ids and titles gives broadest coverage.
+    moved_rooms = false
+    if Room.current.id != nil
+      moved_rooms = @last_room_id != Room.current.id
+    else
+      moved_rooms = @last_room_title != DRRoom.title
+    end
+    if $debug_mode_walkingastro
+      echo "--- room_id=#{Room.current.id}"
+      echo "--- room_title=#{DRRoom.title}"
+      echo "--- last_room_id=#{@last_room_id}"
+      echo "--- last_room_title=#{@last_room_title}"
+      echo "--- moved_rooms=#{moved_rooms}"
+    end
+    return moved_rooms
+  end
+
+  def can_see_sky?
+    # If we didn't see the sky the last time we checked
+    # and we're in the same room then don't redundantly check
+    # so that we don't spam the game window with weather checks.
+    if moved_rooms?
+      echo "Moved rooms, checking if we can see sky in this room" if $debug_mode_walkingastro
+      @can_see_sky = DRC.can_see_sky?
+      @last_room_id = Room.current.id
+      @last_room_title = DRRoom.title
+    else
+      echo "In same room, returning previous answer if we can see the sky: #{@can_see_sky}"
+    end
+    return @can_see_sky
+  end
```

Comment:
If we've moved rooms since the last time we checked if we can see the sky, then check if we can see the sky. Otherwise, just return the previous answer.